crates/oxc_linter/src/rules/unicorn/prefer_date_now.rs

```diff
@@ -217,5 +223,12 @@ fn test() {
         r"function foo(){return-new Date}",
     ];
 
-    Tester::new(PreferDateNow::NAME, pass, fail).test_and_snapshot();
+    let fix = vec![
+        ("new Date().getTime()", "Date.now()"),
+        ("new Date().valueOf()", "Date.now()"),
+        ("Number(new Date())", "Date.now()"),
+        ("BigInt(new Date())", "Date.now()"),
```

Comment:
Can you please add tests that use semicolons, parenthesis, and type casts as well?

Reply:
done, let me know if there're any more cases you want me to add